docs/reference/plugins/initrd_plugin.rst

```diff
@@ -0,0 +1,171 @@
+.. _reference-initrd-plugin:
+
+Initrd plugin
+==============
+
+The Initrd plugin simplifies the building of initrds used by kernel snaps.
+It is intended to generate an initrd for Ubuntu Core systems.
+
+The initrd used by Ubuntu Core is constructed by the ubuntu-core-initramfs tool
+and, while its usage is straightforward, constructing a reliable environment
+where it can be used can be quite cumbersome in the context of cross-building a
+kernel snap. This plugin alleviates those challenges by constructing a minimal
+chroot for the target architecture where any required packages can be added to
+end up in the final initrd.
+
+
+Keys
+----
+
+This plugin provides the following unique keys.
+
+
+initrd-addons
+~~~~~~~~~~~~~
+
+**Type**: list of strings
+
+A list of files to include in the initrd, provided as relative paths to
+``${CRAFT_STAGE}/addons`` like so:
+
+.. code:: yaml
+
+      initrd-addons:
+        - usr/bin/foo
+        - usr/lib/bar
+
+
+initrd-firmware
+~~~~~~~~~~~~~~~
+
+**Type**: list of strings
+
+A list of firmware to include in the initrd, provided as relative paths to
+``${CRAFT_STAGE}/firmware`` like so:
+
+.. code:: yaml
+
+      initrd-firmware:
+        - foo
+        - bar/baz
+
+
+initrd-modules
+~~~~~~~~~~~~~~
+
+**Type**: list of strings
+
+A list of module to include in the initrd, provided as a list of module names.
+If the specified module(s) have dependencies, they are also installed.
+
+
+Core22 and later options
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+For core24 and later, the following options are also supported. However, they
+will only work if the target architecture has a `systemd-boot-efi <https://packages.ubuntu.com/noble/systemd-boot-efi>`_
+package in the archive. Otherwise, the build will fail.
```

Comment:
```suggestion
core22 and higher
~~~~~~~~~~~~~~~~~

For snaps on core22 and higher, the following keys are also available. However, they
will only work if the target architecture has a `systemd-boot-efi <https://packages.ubuntu.com/noble/systemd-boot-efi>`_
package in the archive. Otherwise, the build will fail.
```

Minor, but the metaphorical chronology of core versions and releases gets messy, so it's simpler and more accurate to say "higher/lower". A rule that will come with the upcoming Starcraft style guide.